youtube_dl/extractor/screenjunkies.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import unicode_literals
+
+import re
+import json
+
+from .common import (
+    InfoExtractor,
+    ExtractorError
+)
+
+class ScreenJunkiesIE(InfoExtractor):
+    _VALID_URL = r'http://www.screenjunkies.com/video/(?P<id>.+)'
+
+    _TESTS = [{
+        'url': 'http://www.screenjunkies.com/video/best-quentin-tarantino-movie-2841915',
+        'info_dict': {
+            'id': '2841915',
+            'ext': 'mp4',
+            'title': 'Best Quentin Tarantino Movie',
+        },
+    }, {
+        'url': 'http://www.screenjunkies.com/video/honest-trailers-the-dark-knight',
+        'info_dict': {
+            'id': '2348808',
+            'ext': 'mp4',
+            'title': "Honest Trailers - 'The Dark Knight'",
+        },
+    }, {
+        'url': 'http://www.screenjunkies.com/video/knocking-dead-ep-1-the-show-so-far-3003285',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_link = self._match_id(url)
+        video_id = self._search_regex(r'.*-(\d+)', url, 'id', default=None, fatal=False)
+
+        if not video_id: # Older urls didn't have the id in them, but we can grab it manually
+            webpage = self._download_webpage(url, video_link)
+            video_id = self._search_regex(r'src="/embed/(\d+)"', webpage, 'video id')
+
+        webpage = self._download_webpage('http://www.screenjunkies.com/embed/%s' %video_id, video_id)
+        video_vars_str = self._html_search_regex(r'embedVars = (\{.+\})\s*</script>', webpage, 'video variables', flags=re.S)
+        video_vars = json.loads(video_vars_str)
+
+        # TODO: Figure out required cookies
+        if video_vars['subscriptionLevel'] > 0:
+            raise ExtractorError('This video requires ScreenJunkiesPlus')
+
+        formats = []
+        for f in video_vars['media']:
+            if not f['mediaPurpose'] == 'play':
+                continue
+
+            formats.append({
+                'url': f['uri'],
+                'tbr': f['bitRate'],
+                'width': f['width'],
+                'height': f['height'],
+                'format': 'mp4',
+            })
+        formats.sort(key=lambda d: d['tbr'])
+
+        return {
+            'id': video_id,
+            'title': video_vars['contentName'],
+            'thumbnail': video_vars['thumbUri'],
```

Comment:
Extraction should be tolerate to missing fields. Everything apart from title and formats considered optional and should not break the extraction if missing.